logstash-core/lib/logstash/webserver.rb

```diff
@@ -8,19 +8,20 @@
 module LogStash 
   class WebServer
 
+    class HostBindingError < StandardError; end
+
     extend Forwardable
 
     attr_reader :logger, :status, :config, :options, :cli_options, :runner, :binder, :events
 
     def_delegator :@runner, :stats
 
-    DEFAULT_HOST = "127.0.0.1".freeze
```

Comment:
I'd also prefer that the bind-fail-next-bind loop was done with puma itself instead of looping before and being vulnerable to a race condition.

it should be simple to wrap the puma init code in an rescue/retry block, no?